### ISSUE TITLE
Cannot reproduce the result in tab2 of scannet 3d qa

### ISSUE THREAD

**Issue by Hoyyyaard** [Open3DA/LL3DA#4]
**I run   bash scripts/opt-1.3b/tuning.scanqa.sh and get results:**

[BLEU-1] Mean: 0.0028, Max: 0.2000, Min: 0.0000
[BLEU-2] Mean: 0.0004, Max: 0.1195, Min: 0.0000
[BLEU-3] Mean: 0.0000, Max: 0.0000, Min: 0.0000
[BLEU-4] Mean: 0.0000, Max: 0.0000, Min: 0.0000
[CIDEr] Mean: 0.0003, Max: 0.0786, Min: 0.0000
[ROUGE-L] Mean: 0.0045, Max: 0.2919, Min: 0.0000
[METEOR] Mean: 0.0027, Max: 0.1846, Min: 0.0000

**terminal output ：**
kept 71 scans out of 312
kept 4675 annotations in 71 scans...
Some weights of InstructBlipQFormerModel were not initialized from the model checkpoint at bert-base-embedding and are newly initialized: ['encoder.layer.0.attention.attention.key.bias', 'encoder.layer.0.attention.attention.key.weight', 'encoder.layer.0.attention.attention.query.bias', 'encoder.layer.0.attention.attention.query.weight', 'encoder.layer.0.attention.attention.value.bias', 'encoder.layer.0.attention.attention.value.weight', 'encoder.layer.0.attention.output.LayerNorm.bias', 'encoder.layer.0.attention.output.LayerNorm.weight', 'encoder.layer.0.attention.output.dense.bias', 'encoder.layer.0.attention.output.dense.weight', 'encoder.layer.0.crossattention.attention.key.bias', 'encoder.layer.0.crossattention.attention.key.weight', 'encoder.layer.0.crossattention.attention.query.bias', 'encoder.layer.0.crossattention.attention.query.weight', 'encoder.layer.0.crossattention.attention.value.bias', 'encoder.layer.0.crossattention.attention.value.weight', 'encoder.layer.0.crossattention.output.LayerNorm.bias', 'encoder.layer.0.crossattention.output.LayerNorm.weight', 'encoder.layer.0.crossattention.output.dense.bias', 'encoder.layer.0.crossattention.output.dense.weight', 'encoder.layer.0.intermediate.dense.bias', 'encoder.layer.0.intermediate.dense.weight', 'encoder.layer.0.intermediate_query.dense.bias', 'encoder.layer.0.intermediate_query.dense.weight', 'encoder.layer.0.output.LayerNorm.bias', 'encoder.layer.0.output.LayerNorm.weight', 'encoder.layer.0.output.dense.bias', 'encoder.layer.0.output.dense.weight', 'encoder.layer.0.output_query.LayerNorm.bias', 'encoder.layer.0.output_query.LayerNorm.weight', 'encoder.layer.0.output_query.dense.bias', 'encoder.layer.0.output_query.dense.weight', 'encoder.layer.1.attention.attention.key.bias', 'encoder.layer.1.attention.attention.key.weight', 'encoder.layer.1.attention.attention.query.bias', 'encoder.layer.1.attention.attention.query.weight', 'encoder.layer.1.attention.attention.value.bias', 'encoder.layer.1.attention.attention.value.weight', 'encoder.layer.1.attention.output.LayerNorm.bias', 'encoder.layer.1.attention.output.LayerNorm.weight', 'encoder.layer.1.attention.output.dense.bias', 'encoder.layer.1.attention.output.dense.weight', 'encoder.layer.1.intermediate.dense.bias', 'encoder.layer.1.intermediate.dense.weight', 'encoder.layer.1.intermediate_query.dense.bias', 'encoder.layer.1.intermediate_query.dense.weight', 'encoder.layer.1.output.LayerNorm.bias', 'encoder.layer.1.output.LayerNorm.weight', 'encoder.layer.1.output.dense.bias', 'encoder.layer.1.output.dense.weight', 'encoder.layer.1.output_query.LayerNorm.bias', 'encoder.layer.1.output_query.LayerNorm.weight', 'encoder.layer.1.output_query.dense.bias', 'encoder.layer.1.output_query.dense.weight', 'encoder.layer.2.attention.attention.key.bias', 'encoder.layer.2.attention.attention.key.weight', 'encoder.layer.2.attention.attention.query.bias', 'encoder.layer.2.attention.attention.query.weight', 'encoder.layer.2.attention.attention.value.bias', 'encoder.layer.2.attention.attention.value.weight', 'encoder.layer.2.attention.output.LayerNorm.bias', 'encoder.layer.2.attention.output.LayerNorm.weight', 'encoder.layer.2.attention.output.dense.bias', 'encoder.layer.2.attention.output.dense.weight', 'encoder.layer.2.crossattention.attention.key.bias', 'encoder.layer.2.crossattention.attention.key.weight', 'encoder.layer.2.crossattention.attention.query.bias', 'encoder.layer.2.crossattention.attention.query.weight', 'encoder.layer.2.crossattention.attention.value.bias', 'encoder.layer.2.crossattention.attention.value.weight', 'encoder.layer.2.crossattention.output.LayerNorm.bias', 'encoder.layer.2.crossattention.output.LayerNorm.weight', 'encoder.layer.2.crossattention.output.dense.bias', 'encoder.layer.2.crossattention.output.dense.weight', 'encoder.layer.2.intermediate.dense.bias', 'encoder.layer.2.intermediate.dense.weight', 'encoder.layer.2.intermediate_query.dense.bias', 'encoder.layer.2.intermediate_query.dense.weight', 'encoder.layer.2.output.LayerNorm.bias', 'encoder.layer.2.output.LayerNorm.weight', 'encoder.layer.2.output.dense.bias', 'encoder.layer.2.output.dense.weight', 'encoder.layer.2.output_query.LayerNorm.bias', 'encoder.layer.2.output_query.LayerNorm.weight', 'encoder.layer.2.output_query.dense.bias', 'encoder.layer.2.output_query.dense.weight', 'encoder.layer.3.attention.attention.key.bias', 'encoder.layer.3.attention.attention.key.weight', 'encoder.layer.3.attention.attention.query.bias', 'encoder.layer.3.attention.attention.query.weight', 'encoder.layer.3.attention.attention.value.bias', 'encoder.layer.3.attention.attention.value.weight', 'encoder.layer.3.attention.output.LayerNorm.bias', 'encoder.layer.3.attention.output.LayerNorm.weight', 'encoder.layer.3.attention.output.dense.bias', 'encoder.layer.3.attention.output.dense.weight', 'encoder.layer.3.intermediate.dense.bias', 'encoder.layer.3.intermediate.dense.weight', 'encoder.layer.3.intermediate_query.dense.bias', 'encoder.layer.3.intermediate_query.dense.weight', 'encoder.layer.3.output.LayerNorm.bias', 'encoder.layer.3.output.LayerNorm.weight', 'encoder.layer.3.output.dense.bias', 'encoder.layer.3.output.dense.weight', 'encoder.layer.3.output_query.LayerNorm.bias', 'encoder.layer.3.output_query.LayerNorm.weight', 'encoder.layer.3.output_query.dense.bias', 'encoder.layer.3.output_query.dense.weight', 'encoder.layer.4.attention.attention.key.bias', 'encoder.layer.4.attention.attention.key.weight', 'encoder.layer.4.attention.attention.query.bias', 'encoder.layer.4.attention.attention.query.weight', 'encoder.layer.4.attention.attention.value.bias', 'encoder.layer.4.attention.attention.value.weight', 'encoder.layer.4.attention.output.LayerNorm.bias', 'encoder.layer.4.attention.output.LayerNorm.weight', 'encoder.layer.4.attention.output.dense.bias', 'encoder.layer.4.attention.output.dense.weight', 'encoder.layer.4.crossattention.attention.key.bias', 'encoder.layer.4.crossattention.attention.key.weight', 'encoder.layer.4.crossattention.attention.query.bias', 'encoder.layer.4.crossattention.attention.query.weight', 'encoder.layer.4.crossattention.attention.value.bias', 'encoder.layer.4.crossattention.attention.value.weight', 'encoder.layer.4.crossattention.output.LayerNorm.bias', 'encoder.layer.4.crossattention.output.LayerNorm.weight', 'encoder.layer.4.crossattention.output.dense.bias', 'encoder.layer.4.crossattention.output.dense.weight', 'encoder.layer.4.intermediate.dense.bias', 'encoder.layer.4.intermediate.dense.weight', 'encoder.layer.4.intermediate_query.dense.bias', 'encoder.layer.4.intermediate_query.dense.weight', 'encoder.layer.4.output.LayerNorm.bias', 'encoder.layer.4.output.LayerNorm.weight', 'encoder.layer.4.output.dense.bias', 'encoder.layer.4.output.dense.weight', 'encoder.layer.4.output_query.LayerNorm.bias', 'encoder.layer.4.output_query.LayerNorm.weight', 'encoder.layer.4.output_query.dense.bias', 'encoder.layer.4.output_query.dense.weight', 'encoder.layer.5.attention.attention.key.bias', 'encoder.layer.5.attention.attention.key.weight', 'encoder.layer.5.attention.attention.query.bias', 'encoder.layer.5.attention.attention.query.weight', 'encoder.layer.5.attention.attention.value.bias', 'encoder.layer.5.attention.attention.value.weight', 'encoder.layer.5.attention.output.LayerNorm.bias', 'encoder.layer.5.attention.output.LayerNorm.weight', 'encoder.layer.5.attention.output.dense.bias', 'encoder.layer.5.attention.output.dense.weight', 'encoder.layer.5.intermediate.dense.bias', 'encoder.layer.5.intermediate.dense.weight', 'encoder.layer.5.intermediate_query.dense.bias', 'encoder.layer.5.intermediate_query.dense.weight', 'encoder.layer.5.output.LayerNorm.bias', 'encoder.layer.5.output.LayerNorm.weight', 'encoder.layer.5.output.dense.bias', 'encoder.layer.5.output.dense.weight', 'encoder.layer.5.output_query.LayerNorm.bias', 'encoder.layer.5.output_query.LayerNorm.weight', 'encoder.layer.5.output_query.dense.bias', 'encoder.layer.5.output_query.dense.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
test the model from scratch...
preparing corpus...
/home/admin/micromamba/envs/ll3da/lib/python3.8/site-packages/scipy/__init__.py:143: UserWarning: A NumPy version >=1.19.5 and <1.27.0 is required for this version of SciPy (detected version 1.19.0)
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
/home/admin/micromamba/envs/ll3da/lib/python3.8/site-packages/scipy/__init__.py:143: UserWarning: A NumPy version >=1.19.5 and <1.27.0 is required for this version of SciPy (detected version 1.19.0)
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
/home/admin/micromamba/envs/ll3da/lib/python3.8/site-packages/scipy/__init__.py:143: UserWarning: A NumPy version >=1.19.5 and <1.27.0 is required for this version of SciPy (detected version 1.19.0)
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
/home/admin/micromamba/envs/ll3da/lib/python3.8/site-packages/scipy/__init__.py:143: UserWarning: A NumPy version >=1.19.5 and <1.27.0 is required for this version of SciPy (detected version 1.19.0)
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
Evaluate ; Batch [0/37]; Evaluating on iter: -1; Iter time 32.07; Mem 8703.97MB
Evaluate ; Batch [10/37]; Evaluating on iter: -1; Iter time 6.59; Mem 8703.99MB
Evaluate ; Batch [20/37]; Evaluating on iter: -1; Iter time 6.63; Mem 8703.99MB
Evaluate ; Batch [30/37]; Evaluating on iter: -1; Iter time 6.70; Mem 8703.99MB

----------------------Evaluation-----------------------

[BLEU-1] Mean: 0.0028, Max: 0.2000, Min: 0.0000
[BLEU-2] Mean: 0.0004, Max: 0.1195, Min: 0.0000
[BLEU-3] Mean: 0.0000, Max: 0.0000, Min: 0.0000
[BLEU-4] Mean: 0.0000, Max: 0.0000, Min: 0.0000
[CIDEr] Mean: 0.0003, Max: 0.0786, Min: 0.0000
[ROUGE-L] Mean: 0.0045, Max: 0.2919, Min: 0.0000
[METEOR] Mean: 0.0027, Max: 0.1846, Min: 0.0000

**Comment by ch3cook-fdu** [comment]
The results seem way off. Have you run the `bash scripts/opt-1.3b/train.generalist.sh`?

**Comment by Hoyyyaard** [comment]
I'm sorry I don't have all the data sets yet so I didn't run `bash scripts/opt-1.3b/train.generalist.sh?`.
Whether the warning that occurred while loading the model is normal?
I run `bash scripts/opt-1.3b/tuning.scanqa.sh` again and get  similar results :

[BLEU-1] Mean: 0.0082, Max: 0.2000, Min: 0.0000
[BLEU-2] Mean: 0.0010, Max: 0.1690, Min: 0.0000
[BLEU-3] Mean: 0.0003, Max: 0.1300, Min: 0.0000
[BLEU-4] Mean: 0.0000, Max: 0.0000, Min: 0.0000
[CIDEr] Mean: 0.0002, Max: 0.0854, Min: 0.0000
[ROUGE-L] Mean: 0.0077, Max: 0.3297, Min: 0.0000
[METEOR] Mean: 0.0029, Max: 0.1980, Min: 0.0000



Here is my env:
certifi            2024.2.2
charset-normalizer 3.3.2
Cython             3.0.9
decorator          5.1.1
filelock           3.13.1
fsspec             2024.2.0
h5py               3.10.0
huggingface-hub    0.21.4
idna               3.6
networkx           2.2
numpy              1.19.0
packaging          23.2
pillow             10.2.0
pip                24.0
plyfile            1.0.3
pointnet2          0.0.0
PyYAML             6.0.1
regex              2023.12.25
requests           2.31.0
safetensors        0.4.2
scipy              1.10.1
setuptools         69.1.1
tokenizers         0.15.2
torch              1.13.1+cu116
torchaudio         0.13.1+cu116
torchvision        0.14.1+cu116
tqdm               4.66.2
transformers       4.37.0
trimesh            2.35.39
typing_extensions  4.10.0
urllib3            2.2.1
wheel              0.42.0

**Comment by ch3cook-fdu** [comment]
I believe as long as you can run the code, the environment should not be a problem.

In fact, I notice the message:

```
test the model from scratch...
preparing corpus...
```

It seems the pre-trained weights are not correctly loaded during inference. Could you check the `--test_ckpt`?

**Comment by Hoyyyaard** [comment]
I cannot find the pretrain file provided by the readme of `./ckpts/opt-1.3b/ll3da-generalist/checkpoint.pth`

**Comment by ch3cook-fdu** [comment]
You can also directly train the model for QA only from scratch:

```{bash}
python main.py \
    --use_color --use_normal \
    --detector detector_Vote2Cap_DETR \
    --captioner ll3da \
    --warm_lr_epochs 0 \
    --dataset unified_scanqa \
    --vocab facebook/opt-1.3b \
    --qformer_vocab bert-base-embedding \
    --checkpoint_dir ./ckpts/opt-1.3b/ll3da-scanqa-tuned \
    --max_epoch 24 \
    --dist_url tcp://localhost:333 \
    --eval_every_iteration 4000 \
    --start_eval_after -1 \
    --save_every 10000 \
    --criterion 'CiDEr' \
    --freeze_detector --freeze_llm \
    --batchsize_per_gpu 8 --ngpus 2 --base_lr 1e-4 --final_lr 1e-6 \
    --max_des_len 224 \
    --max_prompt 1 --use_beam_search
```

Due to the size of the model, we could not upload the pre-trained weights on github for now. We will release the weights in other formats for the 3D generalist.

**Comment by Hoyyyaard** [comment]
Thanks so much!
